Assets/Scripts/UI/InGame/Graphics/ResolutionSwitcher.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Assets.Scripts.UI.InGame
+{
+	public class ResolutionSwitcher : MonoBehaviour {
+
+		public Dropdown dropdown;
+		public Toggle toggle;
+		public Text DropDownLabel;
+
+		public string Resolution { get; set; }
+		public int ScreenMode { get; set; }
+
+		private void OnEnable() {
+			if(Screen.fullScreen == true)
+			{
+				toggle.isOn = true;
+			}
+			else
+			{
+				toggle.isOn = false;
+			}
+
+			foreach(Resolution resolution in Screen.resolutions)
+			{
+				string resolutionText = resolution.ToString();
+				Dropdown.OptionData op = new Dropdown.OptionData(resolutionText);
+				dropdown.options.Add(op);
+				if(resolution.Equals(Screen.currentResolution))
+				{
+					dropdown.value = dropdown.options.IndexOf(op);
+				}
+				
+			}
+		}
+
+		public void ChangeResolution()
+		{
+			string res = DropDownLabel.text;
+			for(int i = 0; i < Screen.resolutions.Length; i++)
+			{
+				Resolution temp = Screen.resolutions[i];
+				if(temp.ToString().Equals(res))
+				{
+					Screen.SetResolution(temp.width, temp.height, toggle.GetComponent<Toggle>().isOn, temp.refreshRate);
```

Comment:
Once we assign the resolution, we can break from the loop

Reply:
This also loops through all the possible resolutions the user can use and populates the dropdown with them.